### PR TITLE
NW4 - Gulnihal Naldoken - JS core 1 - Week 3

### PR DESCRIPTION
**Pull request by gulnihaln**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Gulnihal
- Your City: Manchester
- Your Slack Name: Gulnihal

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Comment by myaskylar** [comment]
Your codes are clean and readable. Excellent work Gulnihaln!

File: 1-exercises/F-array-forEach/exercise.js [hunk 1]
```diff
@@ -9,6 +9,22 @@
...
+  }else if (number % 3 === 0){
+    return "Fizz";
+  }else {
+    return number;
+  }
+});
+
+console.log(newArr);
+console.log(arr); //<---------- original array
+
```
Comment (myaskylar) [inline 1-exercises/F-array-forEach/exercise.js:27]: I like how you log the original array to test and comment it here. Great job!

Comment (gulnihaln) [reply 1-exercises/F-array-forEach/exercise.js:27]: > Your codes are clean and readable. Excellent work Gulnihaln!

Thanks for the review Mya !

Comment (Maziarmajd) [reply 1-exercises/F-array-forEach/exercise.js:27]: clean codes and great work Gulnihal. I am agree with Mya, great job.

**Comment by gulnihaln** [comment]
@zeinaji I can't see some of my commits here, I've tried to do them again but still doesn't exists. Please can you see all work from my repositories?

File: 1-exercises/F-array-forEach/exercise.js (hunk 1 above)
Comment (gulnihaln) [reply 1-exercises/F-array-forEach/exercise.js:27]: Thanks Maziar !

File: 2-mandatory/5-eligible-students.js [hunk 2]
```diff
@@ -7,7 +7,10 @@
   - Returns an array containing only the names of the who have attended AT LEAST 8 classes
 */
 
-function getEligibleStudents() {}
+function getEligibleStudents(attendanceCounts) {
+  const whoAttendedAtLeast8classes = attendanceCounts.filter(number => number[1] >= 8);
+  return whoAttendedAtLeast8classes.map(names => names.slice(0, 1)).flat();
```
Comment (Morteza-Khojasteh) [inline 2-mandatory/5-eligible-students.js:12]: Well done, a good approach for the .map and arrow functions.

**Comment by zeinaji** [comment]
> @zeinaji I can't see some of my commits here, I've tried to do them again but still doesn't exists. Please can you see all work from my repositories?

Hey Gulnihal, sorry for the late reply I was away, I can see all of exercises and mandatory, what files did the commits include?

**Comment by gulnihaln** [comment]
> > @zeinaji I can't see some of my commits here, I've tried to do them again but still doesn't exists. Please can you see all work from my repositories?
> 
> Hey Gulnihal, sorry for the late reply I was away, I can see all of exercises and mandatory, what files did the commits include?

Hey Zein, no problem. I think they're alright after the last pushing last night.  I don't what happened earlier, I couldn't see some solution even if I tried to push them for a few times.

File: 2-mandatory/8-password-validator.js [hunk 3]
```diff
@@ -24,7 +24,17 @@ PasswordValidationResult=  [false, false, false, false, true]
 
 */
 
-function validatePasswords(passwords) {}
+function validatePasswords(passwords) {
+  const getNewPassword = passwords.map((element, index) => {
```
Comment (zeinaji) [inline 2-mandatory/8-password-validator.js:28]: You could have the `return` on line 36 here instead and get rid of the `return` and curly brackets in the `map` method, as the booleans all form one statement, so it'd be 
```
function validatePasswords(passwords) {
  return passwords.map((element, index) => 
    containsLowercaseLetter(element) &&
           containsUppercaseLetter(element) &&
           containsNumber(element) &&
           containsSymbol(element) &&
           passwords.indexOf(element) === index &&
           element.length >= 5 
)}
```

**Review by zeinaji — COMMENTED** [review]
Great work Gulnihal! Array methods well used and it's all very clear, with variables extracted were possible and meaningful names throughout. Also nice use of arrow functions 🙂

**Pull request closed without merging**